components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts

```diff
@@ -260,9 +261,18 @@ handler.on(PanelActions.approveHardwareTransaction.getType(), async (store: Stor
   await navigateToConnectHardwareWallet(store)
   const apiProxy = getWalletPanelApiProxy()
   if (hardwareAccount.vendor === BraveWallet.LEDGER_HARDWARE_VENDOR) {
-    const { success, error, code } = (found.coin === BraveWallet.CoinType.ETH)
-       ? await signLedgerEthereumTransaction(apiProxy, hardwareAccount.path, txInfo, found.coin)
-       : await signLedgerFilecoinTransaction(apiProxy, txInfo, found.coin)
+    let success, error, code
+    switch (found.coin) {
+      case BraveWallet.CoinType.FIL:
+        ({ success, error, code } = await signLedgerFilecoinTransaction(apiProxy, txInfo, found.coin))
+        break
+      case BraveWallet.CoinType.SOL:
+        ({ success, error, code } = await signLedgerSolanaTransaction(apiProxy, hardwareAccount.path, txInfo, found.coin))
+        break
+      default:
```

Comment:
Does it make since to default to error case here rather than Ethereum and then explicitly match on CoinType.ETH?

Reply:
Yeah, that seems like a reasonable way to handle it to me